sdk/core/core-https/src/xhrHttpsClient.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/// <reference lib="dom" />
+
+import { AbortError } from "@azure/abort-controller";
+import {
+  HttpsClient,
+  PipelineRequest,
+  PipelineResponse,
+  TransferProgressEvent,
+  HttpHeaders
+} from "./interfaces";
+import { RestError } from "./restError";
+import { createHttpHeaders } from "./httpHeaders";
+
+function isReadableStream(body: any): body is NodeJS.ReadableStream {
+  return body && typeof body.pipe === "function";
+}
+
+/**
+ * A HttpsClient implementation that uses XMLHttpRequest to send HTTPS requests.
+ */
+export class XhrHttpsClient implements HttpsClient {
+  /**
+   * Makes a request over an underlying transport layer and returns the response.
+   * @param request The request to be made.
+   */
+  public async sendRequest(request: PipelineRequest): Promise<PipelineResponse> {
+    const xhr = new XMLHttpRequest();
+
+    if (request.proxySettings) {
+      throw new Error("HTTP proxy is not supported in browser environment");
+    }
+
+    const abortSignal = request.abortSignal;
+    if (abortSignal) {
+      if (abortSignal.aborted) {
+        throw new AbortError("The operation was aborted.");
+      }
+
+      const listener = (): void => {
+        xhr.abort();
+      };
+      abortSignal.addEventListener("abort", listener);
+      xhr.addEventListener("readystatechange", () => {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+          abortSignal.removeEventListener("abort", listener);
+        }
+      });
+    }
+
+    addProgressListener(xhr.upload, request.onUploadProgress);
+    addProgressListener(xhr, request.onDownloadProgress);
+
+    if (request.formData) {
```

Comment:
Same comments from the node http client around handling formData in a policy applies here.

Reply:
This one is more do-able to me, though it'd be the first time we had a browser vs a node version of a policy that I am aware of.

Reply:
We do have a policy today that's different in browsers vs node:
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-http/src/policies/disableResponseDecompressionPolicy.browser.ts
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-http/src/policies/disableResponseDecompressionPolicy.ts

Though, if it really doesn't make sense to do it in the browser case, then I'm not sure if we should in the node case either (just so that we can keep the expectations on what an HttpClient expects to see the same across environments).

Reply:
hmm that *is* a tempting thing to do

Reply:
I think I'll look at pulling this out into a policy in my next PR that ports over policies!